CMakeLists.txt

```diff
@@ -3,68 +3,84 @@
 #####################################
 cmake_minimum_required(VERSION 3.15.0 FATAL_ERROR)
 
-# Define the Project Name and Description
-project (crow_all LANGUAGES CXX)
-
-# Define the module path
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake/")
+# Define the project name and language
+project(Crow
+	LANGUAGES CXX
+)
 
-# Set required C++ Standard
+# Set required C++ standard
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
 
-if (NOT CMAKE_BUILD_TYPE)
+# Default to build type "Release"
+if(NOT CMAKE_BUILD_TYPE)
 	message(STATUS "No build type selected, default to Release")
 	set(CMAKE_BUILD_TYPE "Release")
 endif()
 
 #####################################
 # Define Options
 #####################################
-option(BUILD_EXAMPLES "Builds the examples in the project" ON)
-option(BUILD_TESTING  "Builds the tests in the project"    ON)
+option(CROW_BUILD_EXAMPLES      "Build the examples in the project"      ON )
+option(CROW_BUILD_TESTS         "Build the tests in the project"         ON )
+option(CROW_AMALGAMATE          "Combine all headers into one"           OFF)
+option(CROW_INSTALL             "Add install step for Crow"              ON )
 
-#####################################
-# Define CMake Module Imports
-#####################################
-include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/dependencies.cmake)
-include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/compiler_options.cmake)
+option(CROW_FEATURE_SSL         "Enable SSL capabilities (OpenSSL)"      OFF)
+option(CROW_FEATURE_COMPRESSION "Enable compression capabilities (ZLIB)" OFF)
```

Comment:
Might be best to use the existing CROW_ENABLE_COMPRESSION and CROW_ENABLE_SSL compiler definitions